astroquery/heasarc/core.py

```diff
@@ -135,290 +455,293 @@ def query_object_async(self, object_name, mission, *,
-            may also be entered as a string.
-            (adapted from nrao module)
-        mission : str
-            Mission table to search from
-        radius :
-            Astropy Quantity object, or a string that can be parsed into one.
-            e.g., '1 degree' or 1*u.degree.
-        cache : bool
-            Defaults to True. If set overrides global caching behavior.
-            See :ref:`caching documentation <astroquery_cache>`.
-        **kwargs :
-            see `~astroquery.heasarc.HeasarcClass._args_to_payload` for list
-            of additional parameters that can be used to refine search query
+        provider : str
+            Which cloud data provider to use. Currently, only 'aws' is
+            supported.
+        profile : str
+            Profile to use to identify yourself to the cloud provider
+            (usually in ~/.aws/config).
         """
-        # Convert the coordinates to FK5
-        c = commons.parse_coordinates(position).transform_to(coordinates.FK5)
-        kwargs['coordsys'] = 'fk5'
-        kwargs['equinox'] = 2000
-
-        # Generate the request
-        # Fixed string representation of coordinates ensures that request payload
-        # does not depend on python/astropy version for the same input coordinates
-        request_payload = self._args_to_payload(
-            mission=mission,
-            entry=f"{c.ra.degree:.10f},{c.dec.degree:.10f}",
-            radius=u.Quantity(radius),
-            **kwargs
-        )
 
-        # Return payload if requested
-        if get_query_payload:
-            return request_payload
+        try:
+            import boto3
+            import botocore
+        except ImportError:
+            raise ImportError(
+                'The cloud feature requires the boto3 package. '
+                'Install it first.'
+            )
+
+        if profile is None:
+            log.info('Enabling annonymous cloud data access ...')
+            config = botocore.client.Config(
+                signature_version=botocore.UNSIGNED)
+            self.s3_resource = boto3.resource('s3', config=config)
+
+        elif isinstance(profile, bool) and not profile:
+            # profile is False, use system env credentials
+            self.s3_resource = boto3.resource('s3')
 
-        # Submit the request
-        return self.query_async(request_payload, cache=cache)
+        else:
+            log.info('Enabling cloud data access with profile: {profile} ...')
+            session = boto3.session.Session(profile_name=profile)
+            self.s3_resource = session.resource(service_name='s3')
 
-    def _old_w3query_fallback(self, content):
-        # old w3query (such as that used in ISDC) return very strange fits, with all ints
+        self.s3_client = self.s3_resource.meta.client
 
-        fits_content = fits.open(BytesIO(content))
+    def download_data(self, links, host='heasarc', location='.'):
```

Comment:
I wonder how much of this can be put in the baseclass

Reply:
Presumably some can move to baseclass. The way it is written now is specific to heasarc in the way it handles sciserver and cloud.

Reply:
I expect this to change once we finally consolidate the download capabilities of the different modules, but that doesn't happen before wrapping up this PR, so we should go ahead with this for now.